### PR TITLE
Dash

### PR DESCRIPTION
**Pull request by hnq-jan8**
Press "L" to quickly dash forward
Dash Item acquiring is required
(Extra: Reduced maximum falling speed)

### CODE REVIEW

File: Assets/Scripts/Bahavior/Dash.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    [SerializeField] TrailRenderer tr;
+
+    [SerializeField] Transform groundCheck;
+    [SerializeField] float checkRadius;
+    [SerializeField] LayerMask whatIsGround;
+
+    private bool isGrounded;
+
+    // Start is called before the first frame update
+    void Start()
```
Comment (Thecoldgod04) [inline Assets/Scripts/Bahavior/Dash.cs]: Override Start() function from MovementBehavior class and then call the base.Start() function.

File: Assets/Scripts/Bahavior/Dash.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    // Update is called once per frame
+    void Update()
+    {
+        Dashing();
+    }
+
+    protected virtual void Dashing()
+    {
+        if (PauseMenu.gameIsPaused == true) return;
```
Comment (Thecoldgod04) [inline Assets/Scripts/Bahavior/Dash.cs]: 1. Use this statement only for player input
2. Ask UIManager to know if the player is using UI or not for this statement

File: Assets/Scripts/Bahavior/Dash.cs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    //Must-have variables for movements
+    [SerializeField] IDashInput dashInput;
+    [SerializeField] TrailRenderer tr;
+
+    [SerializeField] Transform groundCheck;
+    [SerializeField] float checkRadius;
+    [SerializeField] LayerMask whatIsGround;
+
+    private bool isGrounded;
```
Comment (Thecoldgod04) [inline Assets/Scripts/Bahavior/Dash.cs]: @Thecoldgod04 Add this attribute into the MovementBehavior class

File: Assets/Scripts/Bahavior/Dash.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    [SerializeField] float dashTime;
+    [SerializeField] float dashCoolDown;
+
+    //Must-have variables for movements
+    [SerializeField] IDashInput dashInput;
+    [SerializeField] TrailRenderer tr;
+
+    [SerializeField] Transform groundCheck;
+    [SerializeField] float checkRadius;
+    [SerializeField] LayerMask whatIsGround;
```
Comment (Thecoldgod04) [inline Assets/Scripts/Bahavior/Dash.cs]: @Thecoldgod04 Add this attribute into the MovementBehavior class

**Review by Thecoldgod04 — COMMENTED** [review]
Loonf

File: Assets/Scripts/Bahavior/Dash.cs (hunk 1 above)
Comment (hnq-jan8) [reply Assets/Scripts/Bahavior/Dash.cs:26]: Oge

File: Assets/Scripts/Bahavior/Dash.cs (hunk 2 above)
Comment (hnq-jan8) [reply Assets/Scripts/Bahavior/Dash.cs]: Ố kề

File: Assets/Scripts/Bahavior/Dash.cs (hunk 3 above)
Comment (hnq-jan8) [reply Assets/Scripts/Bahavior/Dash.cs]: Làm đê

**Review by Thecoldgod04 — APPROVED** [review]
Oge

**Pull request merged**